google/spreadsheet/dust-app-script.js

```diff
@@ -0,0 +1,197 @@
+ */
+function DUST(assistantName, prompt, input) {
+  if (Array.isArray(input)) {
+    return wrapWithError("This function can only be run on a single cell.");
+  }
+
+  try {
+    const assistantId = listAssistants(assistantName);
+    if (!assistantId) {
+      return wrapWithError(`Assistant "${assistantName}" not found.`);
+    }
+
+    const content = createConversationAndGetContent(assistantId, prompt, input);
+    if (!content) {
+      return wrapWithError(
+        "Failed to create conversation or retrieve response."
+      );
+    }
+
+    return content;
+  } catch (error) {
+    console.error("Error in DUST function:", error);
+    return wrapWithError(
+      "An unexpected error occurred. Please check the logs."
+    );
+  }
+}
+
+/**
+ * Retrieve the ID of the specified assistant.
+ *
+ * @param {string} assistantName - Name of the assistant to find.
+ * @return {string|null} The assistant ID if found, null otherwise.
+ */
+function listAssistants(assistantName) {
+  const url = `${BASE_URL}/assistant/agent_configurations`;
+  const options = {
+    method: "get",
+    headers: {
+      Authorization: `Bearer ${API_KEY}`,
+    },
+    muteHttpExceptions: true, // Allows handling HTTP errors manually.
+  };
+
+  try {
+    const response = UrlFetchApp.fetch(url, options);
+    const statusCode = response.getResponseCode();
+
+    if (statusCode !== 200) {
+      console.error(`HTTP error: Received status code ${statusCode}`);
+      return null;
+    }
+
+    const responseText = response.getContentText();
+    if (!responseText) {
+      console.error("API response is empty");
+      return null;
+    }
+
+    const assistants = JSON.parse(responseText);
+    if (!assistants || !assistants.agentConfigurations.length) {
+      console.log("No assistants found in the API response");
+      return null;
+    }
+
+    const assistant = assistants.agentConfigurations.find(
+      (a) => a.name.toLowerCase() === assistantName.toLowerCase()
+    );
+    if (!assistant) {
+      console.log(`Assistant "${assistantName}" not found.`);
+      return null;
+    }
+
+    return assistant.sId;
+  } catch (error) {
+    console.error(`Error in listAssistants: ${error}`);
+    return null;
+  }
+}
+
+/**
+ * Create a new conversation with the specified Dust assistant and retrieve the last agent message content.
+ *
+ * @param {string} assistantId - ID of the assistant to interact with.
+ * @param {string} prompt - User's prompt or question.
+ * @param {string} input - Input from the specified cell.
+ * @return {string|null} The content of the assistant's response, or null on failure.
+ */
+function createConversationAndGetContent(assistantId, prompt, input) {
+  const url = `${BASE_URL}/assistant/conversations`;
+  const payload = {
+    message: {
+      content: `${prompt}\nInput: ${input}`,
+      mentions: [{ configurationId: assistantId }],
+      context: {
+        username: "GoogleSheetsUser",
+        timezone: "UTC",
```

Comment:
For timezone I think you can run `Session.getScriptTimeZone();` to get the real one.

Reply:
Nice 🙃!